src/carousel.tsx

```diff
@@ -457,6 +465,20 @@ const BaseCarousel: React.FC<BaseCarouselProps> = ({
 
     const shouldAutoplay = useShouldAutoplay(!!autoplay, carouselRef);
 
+    React.useEffect(() => {
+        if (
+            initialActiveItem &&
```

Comment:
`0` is a valid value. But this logic discards it.

Reply:
you right, will fix it (I was assuming this logic was safe because the initial slide is always 0, but if changing the prop from non zero to zero will not trigger the effect. I'll change this.

Reply:
fixed